### ISSUE TITLE
DatePicker doesn't fires onChange event when needConfirm=false and showTime enabled

### ISSUE THREAD

**Issue by XAOPT** [ant-design/ant-design#50503]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/antd-reproduce-5x-p3pphb?file=demo.tsx)

### Steps to reproduce

Create datepicker with showTime enabled and needConfirm in false mode.

### What is expected?

onChange event fires when user changing date or time

### What is actually happening?

onChange doesn't fires, but value in input is changing

| Environment | Info |
| --- | --- |
| antd | 5.20.2 |
| React | 18.1.0 |
| System | ubuntu |
| Browser | Chrome |

**Comment by frorff** [comment]
+1

**Comment by zombieJ** [comment]
### TL;DR

Use `onCalendarChange` event instead.

### It's by design

This is the expected behavior. The `onChange` event will only be triggered when the user's action has definitively ended. When using DateTimePicker, clicking on the date or time alone cannot be determined as the user has completed their selection. With `needConfirm`, clicking the confirm button signifies submission, while without `needConfirm`, clicking outside the panel is considered submission. If you need to trigger an event whenever the user makes any selection, you can use the `onCalendarChange` event to listen for these changes.
